File: AdditionalRatePolicy.kt
abstract class AdditionalRatePolicy(private val ratePolicies: List<RatePolicy>) : RatePolicy {
    override fun calculateFee(electricity: Electricity): Double {
        val fee = ratePolicies.sumOf { it.calculateFee(electricity) }
        return afterCalculated(fee)
    }

    abstract protected fun afterCalculated(fee: Double): Double
}

File: AdditionalRatePolicy.swift
class AdditionalRatePolicy: RatePolicy {
    let previous: RatePolicy
    
    init(previous: RatePolicy) {
        self.previous = previous
    }
    
    func calculateFee(of usage: Double) -> Double {
        let fee = previous.calculateFee(of: usage)
        return afterCalculation(fee, using: usage)
    }
    
    func afterCalculation(_ fee: Double, using usage: Double) -> Double {
        fatalError("should be overrided")
    }
}
